### DISCUSSION TITLE
Example: parsing a table (resume/CV) from a PDF file

### DISCUSSION THREAD

**Discussion by adrienjoly** [adrienjoly/npm-pdfreader#112] [Examples of use]
![example cv resume parse convert pdf table to text](https://github.com/adrienjoly/npm-pdfreader-example/raw/master/parseTable.png)

Here is the code required to convert this PDF file into a textual table:

```js
var pdfreader = require("pdfreader");

const nbCols = 2;
const cellPadding = 40; // each cell is padded to fit 40 characters
const columnQuantitizer = (item) => parseFloat(item.x) >= 20;

const padColumns = (array, nb) =>
  Array.apply(null, { length: nb }).map((val, i) => array[i] || []);
// .. because map() skips undefined elements

const mergeCells = (cells) =>
  (cells || [])
    .map((cell) => cell.text)
    .join("") // merge cells
    .substr(0, cellPadding)
    .padEnd(cellPadding, " "); // padding

const renderMatrix = (matrix) =>
  (matrix || [])
    .map((row, y) => padColumns(row, nbCols).map(mergeCells).join(" | "))
    .join("\n");

var table = new pdfreader.TableParser();

new pdfreader.PdfReader().parseFileItems(filename, function (err, item) {
  if (!item || item.page) {
    // end of file, or page
    console.log(renderMatrix(table.getMatrix()));
    console.log("PAGE:", item.page);
    table = new pdfreader.TableParser(); // new/clear table for next page
  } else if (item.text) {
    // accumulate text items into rows object, per line
    table.processItem(item, columnQuantitizer(item));
  }
});
```

Fork this example from [parsing a CV/résumé](https://github.com/adrienjoly/npm-pdfreader-example).

**Comment by simshengqin** [comment]
hi, I tested the parseTable.js code in your github repo and it is generating 1 column instead of 2. May I check what is the issue?
![image](https://user-images.githubusercontent.com/54699127/162567330-e2a0c1e4-9938-4aeb-9ee2-2dc6673c0823.png)

**Reply by adrienjoly** [reply]
That's weird...
Can you share the source code of the `parseTable.js` file you're running + tell us what version of node.js and `pdfreader` you're using?

**Comment by valdebenitoalbornoz** [comment]
Good job!

It's posible to parse the document, for get a image inside it?

**Reply by adrienjoly** [reply]
As stated on the project home page (https://github.com/adrienjoly/npm-pdfreader), this lib can only parse textual entries, sorry.
